wrapanapi/systems/virtualcenter.py

```diff
@@ -241,14 +286,169 @@ def _get_resource_pool(self, resource_pool_name=None):
+                                                vm_reloc_spec=vm_reloc_spec)
+
+        # Create the StoragePlaceSpec object that will be passed to the RecommendDatastores method
+        storage_spec = vim.StoragePlacementSpec()
+        # Create a empty Config spec data object
+        config = vim.vm.ConfigSpec()
+        # Specification for moving or copying a VM to a different storage pod
+        pod_spec = vim.StorageDrsPodSelectionSpec(storagePod=datastore)
+
+        storage_spec.type = 'clone'
+        storage_spec.cloneName = destination
+        storage_spec.folder = self._get_loc_of_vm(source_template=source_template,
+                                                  progress_callback=progress_callback)
+        storage_spec.podSelectionSpec = pod_spec
+        storage_spec.vm = source_template
+        storage_spec.cloneSpec = vm_clone_spec
+        storage_spec.configSpec = config
+
+        # Returns a StoragePlacementResult object.
+        # For SDRS-enabled pods, this API is intended to replace RelocateVM_Task and CloneVM_Task
+        # SDRS is required for managing aggregated resources of a datastore cluster
+        result = self.system.content.storageResourceManager.RecommendDatastores(
+            storageSpec=storage_spec)
+
+        if result:
+            key = result.recommendations[0].key
+        else:
+            raise AttributeError(
+                "RecommendDatastore task failed to provide host for  {}".format(destination))
+
+        return self.system.content.storageResourceManager.ApplyStorageDrsRecommendation_Task(
+            key=key)
+
+    def _clone(self, destination, resource_pool=None, datastore=None, power_on=True, sparse=False,
+               template=False, provision_timeout=1800, progress_callback=None,
                allowed_datastores=None, cpu=None, ram=None, relocate=False, host=None, **kwargs):
-        """Clone this template to a VM
+        """
+        Clone this template to a VM
         When relocate is True, relocated (migrated) with VMRelocateSpec instead of being cloned
         Returns a VMWareVirtualMachine object
+        Args:
+            destination (string): Name for cloned virtual machine.
+            resource_pool: The resource pool to which this virtual machine should be attached.
+                Can be a vim.ResourcePool object, vim.ClusterComputeResource object, or string.
+            datastore: Datastore or Datastore Cluster to deploy the VM on. Can be a string,
+                vim.Datastore object (for datastore), or vim.StoragePod object (for datastore
+                cluster)
+            power_on (boolean): Specifies whether or not the new VM should be powered on after
+                creation
+            sparse (boolean): If set to true, sets the transformations on the virtual disk to be
+                sprase. IE, growable virtual disk.
+            template (boolean): Specifies whether or not the new virtual machine should be
```

Comment:
I reckon that's bad name.  as_template or make_template should describe this param better.